boards/arc/nsim/nsim_em.dts

```diff
@@ -8,9 +8,35 @@
 
 #include "nsim_em.dtsi"
 
+#ifndef XCCM_ADDR
```

Comment:
Where are these ifdef defines getting defined?

Reply:
I followed the style of other memory, such as [ICCM and DCCM](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arc/nsim/nsim-ccm-mem.dtsi#:~:text=%23ifndef%20ICCM_ADDR,%23endif). Use #ifdef is to have different definition for special application, just like [arc_sec_application](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/boards/arc_secure_services/nsim_sem_normal.dts#:~:text=%23define%20ICCM_ADDR%2040000,DCCM_SIZE%20DT_SIZE_K(256)).

Reply:
That does not feel like a correct use of dts preprocessor macro, why not delete the existing node and redefine it how it's supposed to be instead? That way you wyould not have to do quirky things like `/* must be before include of nsim_em.dtsi */`. Not saying it should be fixed in this PR but maybe this should avoid copying that pattern (and then that can be fixed separately).

Reply:
removed `#ifndef` to avoid mislead